### DISCUSSION TITLE
Version 0.10.0-beta.1 test - ENDED

### DISCUSSION THREAD

**Discussion by Taitava** [Taitava/obsidian-shellcommands#138] [Announcements]
This beta test contains the following new features:

- [Events: Execute shell commands automatically when something happens (#123)](https://github.com/Taitava/obsidian-shellcommands/issues/123).
- [Output channel: modal (#134)](https://github.com/Taitava/obsidian-shellcommands/issues/134): Allows to direct shell command output to a modal where it can be edited and manually redirected to other channels, i.e. to clipboard, file or status bar.
- [Settings: Add availability information to the list of variables (#132)](https://github.com/Taitava/obsidian-shellcommands/issues/132).

**For bug reports regarding this beta release, please write your comments in this discussion**. Please do not create new discussions or issues. For new ideas/feature requests, please create a new discussion in the ideas category, or if it's related to events, please comment in the [events discussion](https://github.com/Taitava/obsidian-shellcommands/discussions/60).

---

This is not a normal version. It's a beta test version and can be unstable. [See these instructions, if you are willing to participate in beta testing](https://github.com/Taitava/obsidian-shellcommands/discussions/98). Do not use any important vault for testing, because you are not able to convert your data.json settings file back to an earlier version!

<hr /><em>This discussion was created from the release <a href='https://github.com/Taitava/obsidian-shellcommands/releases/tag/0.10.0-beta.1'>0.10.0-beta.1</a>.</em>

---

TODO for myself regarding output channel Modal:
- [x] Focus Caret Position in the Text Field when openning the Modal
- [x] Rename output channel _Modal_ to **_Ask after execution_** or something else (still being discussed)

Edit 2022-02-06: I decided to do the following improvements later, not in 0.10.0:
- ~~Add hotkeys `Ctrl`/`Cmd` + `1`, `2`, `3` etc... to use different output channel buttons. Read more: https://github.com/Taitava/obsidian-shellcommands/discussions/139~~
- ~~Add a `Shift` modifier to the abovementioned number hotkeys to close the modal after using an output channel.~~
- ~~Make `Ctrl`/`Cmd` + clicking the buttons also close the modal.~~

**Comment by Mara-Li** [comment]
Oh, it's a pretty great update!
Some suggestion : 
- Allow to set an obsidian icon for event (file menu & folder)
- Allow to set / change the notification output for stdout (override the normal stdout) 

Thank you for this update ! I love to use your plugin with my publish script :D

**Reply by Taitava** [reply]
> Oh, it's a pretty great update!

Thank you! 😎 

> 
>   * Allow to set an obsidian icon for event (file menu & folder)

Yea, it's been on my mind, I just haven't yet created an issue about it. Defining icons that will be used in the menus will be part of a change where I'll start to use some larger icon library, and will also change some of the icons SC now uses in its settings panel. At that point I'll add the ability to define an icon for each shell command, and that icon will be shown in the menus ([maybe also in the left side panel?](https://github.com/Taitava/obsidian-shellcommands/discussions/25)).

> 
>   * Allow to set / change the notification output for stdout (override the normal stdout)

Sorry, I didn't understand this 🙂 . What do you want to achieve? Something else than [the output channels that are listed here](https://publish.obsidian.md/shellcommands/Output+handling/Output+channels#All+output+channels)?

> 
> 
> Thank you for this update ! I love to use your plugin with my publish script :D

Thank you for your feedback! 🙂 I need to take a look at [your publish script](https://forum.obsidian.md/t/yet-another-free-publish-alternative-yafpa-no-longer-maintened/23608)! 😎

**Reply by Taitava** [reply]
> * Allow to set an obsidian icon for event (file menu & folder)

@Mara-Li The icon feature (#240) is now released in `0.14.0`. 🙂

**Comment by FelipeRearden** [comment]
Hello @Taitava !!!!!

Just to let you know that I’m here (I sent you a DM) and I gonna start the tests today :)

Just give 2/3 days. I’m so excited to see the new features !!!! We have a lot ground to cover :)

Have a great day !!!!!

**Comment by FelipeRearden** [comment]
## Modal Output

01- Focus Caret Position in the Text Field when openning the Modal 

02 - Redirect: Support for Keyboard navigation using TAB to switch between outputs 
- if possible change the `border-color` of the selected output 

03- Redirect: CRTL/CMD+ENTER to select the output that was choose by item 2

**Reply by Taitava** [reply]
> 01- Focus Caret Position in the Text Field when openning the Modal

Yea, this really makes sense! I wonder why I didn't come to think about it.

> 
> 02 - Redirect: Support for Keyboard navigation using TAB to switch between outputs
> 
>   * if possible change the `border-color` of the selected output
> 
> 
> 03- Redirect: CRTL/CMD+ENTER to select the output that was choose by item 2

Thank you for this idea! 🙂 I created a new discussion for it: https://github.com/Taitava/obsidian-shellcommands/discussions/139

Btw. I came to think that I might like to change the name of the output channel - _Modal_ is not so informative:

![kuva](https://user-images.githubusercontent.com/13002142/151813652-30ff49ce-f70a-4497-a943-5580878ed1ea.png)

How about something like **_Ask after execution_** ? Other ideas? 🙂

**Reply by FelipeRearden** [reply]
> Yea, this really makes sense! I wonder why I didn't come to think about it.

:)

> Thank you for this idea! 🙂 I created a new discussion for it: https://github.com/Taitava/obsidian-shellcommands/discussions/139

👀 

> Btw. I came to think that I might like to change the name of the output channel - _Modal_ is not so informative: 
> How about something like **_Ask after execution_** ? Other ideas? 🙂 

I like Modal because it is a term familiar to Obsidian users BUT I like `Ask for Input` because it's the term used on iOS/iPad Shortcuts App 

Whatever you decide will be a good name :) Modal, Ask for Input or Ask for Execution

**Reply by FelipeRearden** [reply]
4- Redirect: CMD/CTRL+Click to Close the Modal After selecting an output

I love that the modal stays open after we click into one of the commands. I just think that for some workflows, would be amazing to have a way to close the modal after selecting one command. 

My suggestion would be CRLT/CMD+click to apply the command and close the modal after :)

From #139
> Another idea !!!!: how about we use CMD/CRTL+1 , CMD/CRTL+2 , CMD/CRTL+3 , CMD/CRTL+4 and CMD/CRTL+5 to trigger each one of the Redirect commands instead of TAB??????

If my idea were feasible we could use CMD/CRTL+SHIFT+1 , CMD/CRTL+SHIFT+2, CMD/CRTL+SHIFT+3, CMD/CRTL+SHIFT+4 and CMD/CRTL+SHIFT+5 to achieve the same result with the keyboard!

**Reply by Taitava** [reply]
> 👀
> 
> > Btw. I came to think that I might like to change the name of the output channel - _Modal_ is not so informative:
> > How about something like **_Ask after execution_** ? Other ideas? 🙂
> 
> I like Modal because it is a term familiar to Obsidian users BUT I like `Ask for Input` because it's the term used on iOS/iPad Shortcuts App

Sorry, I didn't quite understand this _Ask for input_. Did you mean _Ask for output_ / _Ask for output channel_ ? 

I started to think that the term _Modal_ is a bit technical (perhaps?) and more because it doesn't indicate that the modal gives the ability to decide the output channel manually each time.

> 4- Redirect: CMD/CTRL+Click to Close the Modal After selecting an output

Could be nice 👍 .

> If my idea were feasible we could use CMD/CRTL+SHIFT+1 , CMD/CRTL+SHIFT+2, CMD/CRTL+SHIFT+3, CMD/CRTL+SHIFT+4 and CMD/CRTL+SHIFT+5 to achieve the same result with the keyboard!

Yea, it might make sense. Then you would not need to hit `Esc` separately to close the modal, as `Shift` is more close to `Ctrl`/`Cmd` than `Esc` would be.

**Reply by FelipeRearden** [reply]
> Sorry, I didn't quite understand this Ask for input. Did you mean Ask for output / Ask for output channel ?

The Shortcuts App on iOS have an action that open a modal for the user write something. The name that used by Apple is `Ask for Input`

![image](https://user-images.githubusercontent.com/85180696/152350823-07eb87a6-c1b8-445b-9e25-4e09c5376ba8.jpeg)

![image](https://user-images.githubusercontent.com/85180696/152350967-58b37690-fb2a-481b-97a6-d83d22511408.jpeg)

> I started to think that the term Modal is a bit technical (perhaps?) and more because it doesn't indicate that the modal gives the ability to decide the output channel manually each time.

100% agree @Taitava !

**Reply by Taitava** [reply]
But I think this is a different thing, it's not so much about _input_, it's about _output_ - with the possibility to modify the output text if wanted.

**Reply by FelipeRearden** [reply]
> But I think this is a different thing, it's not so much about _input_, it's about _output_ - with the possibility to modify the output text if wanted.

Yeah, make more sense!!!!!

**Comment by FelipeRearden** [comment]
Hello @Taitava !!!!!!!

## Version 0.10.0-beta.1

From my tests everything is working perfect !!!!!!

> Events: Execute shell commands automatically when something happens (#123).

I tested everything, from the commands, to the interface and the new variables. I´s absolutely fantastic. 

The notifications for the variables like {{event_tags}} and {{events_yaml_value}} are something that let us know that you care about everything :) 

> Output channel: modal (#134):.

Amazing new feature ! working prefect for every workflow that I tested.

> Settings: Add availability information to the list of variables (#132).

Fantastic compliment  for the variables autocomplete. I cant live without this anymore :)


What a fantastic update we have :)

Have a great day!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Reply by Taitava** [reply]
Thank you, really nice to hear! 😎

**Comment by Taitava** [comment]
I'll release `0.10.0` today, so the beta test will now end.
